core/domain/draft_upgrade_services.py

```diff
@@ -808,13 +843,16 @@ def _convert_states_v30_dict_to_v31_dict(
             if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
                     change.property_name ==
                     exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS):
-                # Get the language code to access the language code correctly.
-                # Ruling out the possibility of any other type for mypy type
-                # checking.
-                assert isinstance(change.new_value, dict)
-                new_voiceovers_mapping: Dict[
-                    str, Dict[str, state_domain.VoiceoverDict]
-                ] = change.new_value['voiceovers_mapping']
+                # Here we use cast because this 'if' condition forces change to
+                # have type EditExpStatePropertyRecordedVoiceoversCmd.
+                edit_recorded_voiceovers_cmd = cast(
+                    exp_domain.EditExpStatePropertyRecordedVoiceoversCmd,
+                    change
+                )
+                recorded_voiceovers = edit_recorded_voiceovers_cmd.new_value
```

Comment:
Is this a dict or a domain object? If the former, end the name with _dict.

Reply:
It's a dict name is changed !!